library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -82,13 +84,41 @@ def self.configured?
       @configured = out["exit"] == 0
     end
 
-    # Creates a new 'single' snapshot
+    # Returns whether creating the given snapshot type is allowed
+    # Information is taken from Linuxrc (DISABLE_SNAPSHOTS)
+    #   * "all" - all snapshot types are temporarily disabled
+    #   * "around" - before and after calling YaST
+    #   * "single" - single snapshot at a given point
+    #
+    # @param [Symbol] one of :around (for :post and :pre snapshots) or :single
+    # @return [Boolean] if snapshot should be created
+    def self.create_snapshot?(snapshot_type)
+      disable_snapshots = Yast::Linuxrc.get_value(Yast::LinuxrcClass::DISABLE_SNAPSHOTS)
+
+      # Feature is not defined on Linuxrc commandline
+      return true if disable_snapshots.nil? || disable_snapshots.empty?
+
+      disable_snapshots = disable_snapshots.downcase.tr("-_.", "").split(",").map(&:to_sym)
```

Comment:
yes, sounds reasonable, thats why I suggest it above, but its usage is spread over all modules, so hard to do bigger changes, maybe only add new API, but old will live with us for long time